django/db/models/sql/query.py

```diff
@@ -303,6 +307,8 @@ def clone(self, klass=None, memo=None, **kwargs):
         # used.
         obj._annotation_select_cache = None
         obj.max_depth = self.max_depth
+        obj.combinator = self.combinator
+        obj.combinator_querysets = list(map(lambda qs: qs.all(), self.combinator_querysets))
```

Comment:
Again, the `.all()` seems to be unnecessary, and will add a lot of overhead on combined QS's.